lib/api/controllers/baseController.js

```diff
@@ -338,8 +314,48 @@ class BaseController {
     const limit = this._kuzzle.config.limits.documentsFetchCount;
 
     if (asked > limit) {
-      errorsManager.throw('search_page_size', limit);
+      errorsManager.throw('services', 'storage', 'get_limit_exceeded');
+    }
+  }
+
+  _cast (name, value, type) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this function to always return the same type. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3800)